src/store.js

```diff
@@ -33,6 +33,9 @@ const store = new Vuex.Store({
     setLoading(state, loading) {
       state.isLoading = loading;
     },
+    setMessage(state, message) {
+      Vue.set(state.messages, message.index, message.message);
```

Comment:
I would prefer you only work with messageId instead of index. Just use messageId everywhere and here when you need the index, use the method `findIndex`